### PR TITLE
Replace NextJS with `Vite`

### PR DESCRIPTION
**Pull request by Da-Colon**
- remove next config and types
- Add vite and react-router-dom
- Add vite config
- Create index.html (update with metadata from next/head)
- Create main.tsx (entry point)
- Add CSS imports, providers to entry point
- Create router.tsx to handle page router
- Add DAOController and Layout element to Router
- Add Pages to Router (A few index -> default component name file changes)
- Replace next's Script with custom hook
- Replace next Link Component with React Router Link
- Replace process.env -> import.meta with update ENV names
- Replace next/image component with Chakra Image component
- Replace next/router push instances with `useNavigate`
- Replace package.json next scripts with vite runner

## Testing
Note: Deployed preview will not work.

I deployed this branch manually on netlify. Here is the link: https://test-vite-fractalframework.netlify.app/

### CODE REVIEW

File: src/router.tsx
Comment (Da-Colon) [inline src/router.tsx:1]: This was surprisingly pretty straight forward to create. Just takes the 'index' of each page folder formally the `page.tsx` and place as the element.

File: .env
Comment (Da-Colon) [inline .env:1]: Don't forget to update local env files as well from `NEXT_PUBLIC_` -> `VITE_APP_`

File: src/hooks/utils/useETHLizardsScript.tsx
Comment (Da-Colon) [inline src/hooks/utils/useETHLizardsScript.tsx:1]: I cannot test this since Goerli is no longer supported. Should we just remove?

File: vite.config.ts
Comment (Da-Colon) [inline vite.config.ts:1]: So far in my testing, all I've had to update here is the plugin for nodepolyfils. This was fixing a 'process is not defined' and 'global is not defined' errors when setting up

**Review by Da-Colon — COMMENTED** [review]
Test everything workflow you can think of. Should only notice the app being smoother everywhere. report any weirdness and I'll look into it.

File: src/components/pages/DaoSettings/components/Metadata/index.tsx [hunk 1]
```diff
@@ -11,13 +10,14 @@ import { useFractal } from '../../../../../providers/App/AppProvider';
 import { ProposalExecuteData } from '../../../../../types';
 import { couldBeENS } from '../../../../../utils/url';
 import { InputComponent } from '../../../../ui/forms/InputComponent';
+import { useNavigate } from 'react-router-dom';
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/components/pages/DaoSettings/components/Metadata/index.tsx:13]: o weird, My IDE should do it on save. but i'll run the link:fix locally

File: src/pages/daos/[daoAddress]/proposal-templates/new/index.tsx [hunk 2]
```diff
@@ -37,8 +37,8 @@ export default function CreateProposalTemplatePage() {
   const [initialProposalTemplate, setInitialProposalTemplate] = useState(DEFAULT_PROPOSAL_TEMPLATE);
   const { t } = useTranslation(['proposalTemplate', 'proposal']);
 
-  const { push } = useRouter();
-  const searchParams = useSearchParams();
+  const navigate = useNavigate();
+  let [searchParams] = useSearchParams();
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/pages/daos/[daoAddress]/proposal-templates/new/index.tsx:41]: docs shows let? https://reactrouter.com/en/main/hooks/use-search-params.

File: src/components/pages/DaoSettings/components/Metadata/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/components/pages/DaoSettings/components/Metadata/index.tsx:13]: Ah it was broken as it was still using a next dependency good catch

**Comment by Da-Colon** [comment]
@mudrila All your comments have been address, sir. (+some)

File: src/vite-env.d.ts
Comment (adamgall) [inline src/vite-env.d.ts:1]: At first I was like "we're missing entries for `TOKEN_PRICE_CACHE_INTERVAL_MINUTES` and `COINGECKO_API_KEY`, but actually those are probably not in here on purpose?? Does the netlify function live outside of Next/Vite?

File: index.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      property="og:description"
+      content="Are you outgrowing your Multisig? Fractal extends Safe treasuries into on-chain hierarchies of permissions, token flows, and governance."
+    />
+    <meta
+      property="og:type"
+      content="website"
+    />
+    <meta
+      property="og:title"
+      content={APP_NAME}
```
Comment (adamgall) [inline index.html:44]: Can we use `VITE_APP_NAME` here?

File: index.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    />
+    <meta
+      property="og:image"
+      content="https://app.fractalframework.xyz/images/hero_image.png"
+    />
+    <meta
+      property="og:image:alt"
+      content={APP_NAME}
```
Comment (adamgall) [inline index.html:52]: Can we use `VITE_APP_NAME` here?

File: index.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    />
+    <meta
+      property="og:site_name"
+      content={APP_NAME}
```
Comment (adamgall) [inline index.html:56]: Can we use `VITE_APP_NAME` here?

File: src/pages/DAOController.tsx [hunk 6]
```diff
@@ -1,10 +1,8 @@
-'use client';
-
 import { Button, Center, Text, VStack, ChakraProvider, extendTheme } from '@chakra-ui/react';
 import { theme } from '@decent-org/fractal-ui';
-import Script from 'next/script';
 import { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
+import { Outlet } from 'react-router-dom';
 import { useNetwork } from 'wagmi';
 import { APP_NAME } from '../constants/common';
```
Comment (adamgall) [inline src/pages/DAOController.tsx:7]: Can we remove this import?

File: src/pages/DAOController.tsx [hunk 7]
```diff
@@ -109,30 +107,14 @@ export default function DAOController({ children }: { children: React.ReactNode
...
-      display = children;
+      display = <Outlet />;
     } else {
       display = <InvalidSafe />;
     }
   }
-
   return (
     <>
       <title>{node?.daoName ? `${node.daoName} | ${APP_NAME}` : APP_NAME}</title>
```
Comment (adamgall) [inline src/pages/DAOController.tsx:117]: Can we use `VITE_APP_NAME` here?

File: src/providers/NetworkConfig/rainbow-kit.config.ts [hunk 8]
```diff
@@ -46,16 +46,16 @@ export const { chains, publicClient } = configureChains(supportedWagmiChains, [
 
 const defaultWallets = [injectedWallet({ chains }), coinbaseWallet({ appName: APP_NAME, chains })];
```
Comment (adamgall) [inline src/providers/NetworkConfig/rainbow-kit.config.ts:47]: Can we use `VITE_APP_NAME` here?

**Review by adamgall — CHANGES_REQUESTED** [review]
Can we use the new `VITE_APP_NAME` env var in all the places in the codebase that we were originally using the `APP_NAME` constant defined in `src/constants/common.ts`? And then, remove that constant from `common.ts`?

**Review by adamgall — CHANGES_REQUESTED** [review]
The deployed site doesn't load if you refresh the page on any non-root URL.

This is known and expected.

There's a simple fix for Netlify hosting, which allows us to keep the `BrowserRouter` and not need to switch to a `HashRouter`.

See this commit from the `fractal-interface-2` repo: https://github.com/decent-dao/fractal-interface-2/commit/6bc770221a739d91e5b7ea052f48b4e7d8cf819a

File: src/vite-env.d.ts
Comment (Da-Colon) [reply src/vite-env.d.ts:1]: thats correct, they live outside of it

File: index.html (hunk 4 above)
Comment (Da-Colon) [reply index.html:52]: Good call on it

**Comment by Da-Colon** [comment]
@adamgall added your updates. I also updated the link to deployed test site (updated PR description too). https://test-vite-fractalframework.netlify.app/

File: global.d.ts
Comment (adamgall) [inline global.d.ts:1]: Do we still need this, considering https://github.com/decent-dao/fractal-interface/pull/1431#discussion_r1520138239

Comment (Da-Colon) [reply global.d.ts:1]: good call...deleted

File: .gitignore
Comment (adamgall) [inline .gitignore:1]: Could probably delete the `next` specific stuff out of here. Also, please move that `./dist` line up into the `# production` section of this file (and you don't need the `.` in `./dist`)

**Review by adamgall — CHANGES_REQUESTED** [review]

File: package.json [hunk 9]
```diff
@@ -37,26 +37,29 @@
...
     "remark-gfm": "^4.0.0",
+    "vite-plugin-node-polyfills": "^0.21.0",
     "viem": "^1.21",
+    "vite": "^5.1.0",
     "wagmi": "^1.4.11",
     "yup": "^1"
   },
   "scripts": {
-    "lint": "eslint . --ext .ts,.tsx",
+    "lint": "eslint . --ext .ts,.tsx --fix",
```
Comment (adamgall) [inline package.json:56]: Please create a separate script for this. Right now there's a Github Action that runs this command, and we don't want that runner to be making any changes to the repo. It should just be succeeding or failing based on whether this script exits with a error or not.

**Review by adamgall — CHANGES_REQUESTED** [review]

File: index.html [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+      href="/favicon.ico"
+    />
+
+    <meta
+      name="theme-color"
+      content="#181716"
+    />
+    <meta
+      name="description"
+      content="Are you outgrowing your Multisig? Fractal extends Safe treasuries into on-chain hierarchies of permissions, token flows, and governance."
```
Comment (adamgall) [inline index.html:16]: Blah, I think that either this should be in an env var, or we just hardcode "Fractal" back into the places where we are using VITE_APP_NAME.

Why have some of this metadata hardcoded and others controlled by environment?

But whatever, it's fine for now.

File: vite.config.mts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import react from '@vitejs/plugin-react-swc';
+import { defineConfig } from 'vite';
+import { nodePolyfills } from 'vite-plugin-node-polyfills';
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [nodePolyfills(), react()],
+  server: {
+    port: 3000,
```
Comment (adamgall) [inline vite.config.mts:8]: @Da-Colon just wondering, does it actually matter to force this to run on 3000? THIS AIN'T A NEXT APP ANY LONGER BOYYEEEEEE

Comment (Da-Colon) [reply vite.config.mts:8]: lol fair. more for me and my history autocomplete :D

File: src/metadata/lizzardsDAO/index.ts
Comment (adamgall) [inline src/metadata/lizzardsDAO/index.ts:1]: Does this whole file and associated other lizard-specific files get removed too? I'm not sure where we stand on that....

File: src/pages/daos/[daoAddress]/proposal-templates/new/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamgall) [reply src/pages/daos/[daoAddress]/proposal-templates/new/index.tsx:38]: Yeah but, please make it a `const` cause we're not changing it.

**Review by adamgall — CHANGES_REQUESTED** [review]

File: src/metadata/lizzardsDAO/index.ts
Comment (Da-Colon) [reply src/metadata/lizzardsDAO/index.ts:1]: @mudrila @tomstuart123 ^^^^

**Review by adamgall — APPROVED** [review]

File: vite.config.mts (hunk 11 above)
Comment (Da-Colon) [reply vite.config.mts:8]: I don't really have a preference. btw we can definately just go with the defaults of vite.

File: src/metadata/lizzardsDAO/index.ts
Comment (adamgall) [reply src/metadata/lizzardsDAO/index.ts:1]: The export from this file isn't being consumed anywhere, so from a codebase perspective, it's safe to delete this file. I wonder if we should hold on to it for future use though...

File: vite.config.mts (hunk 11 above)
Comment (adamgall) [reply vite.config.mts:8]: My preference is that we just go with the vite default

**Review by adamgall — APPROVED** [review]

**Comment by adamgall** [comment]
@Da-Colon can you explain the difference between `chakra`'s `<Link>` component and `react-router-dom`'s `<Link>` component? There seems to be extensive use of both in the codebase, and I'm not sure what the distinction is between them. 

This commit has me scratching my head... why change the `href` to `to` on _these_ `chakra <Link>`'s and not _others_?

**Comment by Da-Colon** [comment]
> @Da-Colon can you explain the difference between `chakra`'s `<Link>` component and `react-router-dom`'s `<Link>` component? There seems to be extensive use of both in the codebase, and I'm not sure what the distinction is between them.
> 
> This commit has me scratching my head... why change the `href` to `to` on _these_ `chakra <Link>`'s and not _others_?

@adamgall 

Chakra-UI's Link basically is a stylable `<a>` element
React-Router's Link adds React Router's functionality to the `<a>` element. https://reactrouter.com/en/main/components/link

image

Switching to `to` property for these allowed it to pick up that we are just navigating to a different page rather than reloading the document

---

As far as the various uses. Looking through how I've updated it.

React-Router's `Link` is used when the navigation is internal

Chakra-UI's `Link` is used when navigation is external or needs to be styled.

In the case of needing to be styled and internal. Chakra's `as` property is used. This allows Chakra's styling ability to be used along side React-Router's properties (why you saw these switch to `to` property as well)

**Comment by tomstuart123** [comment]
So I think this is good to go

Successful Test cases for both multisig + token dao
- Create a DAO 
- delegate (token only)
- vote / sign
- wait for timelock
- execute

these were for subdao, token send and create a proposal flow

just a note - I did have old issues reccur with subdao activity but david pointed me to a seperate PR for that - https://discord.com/channels/839548879216181309/1174434485043998860/1219698481091055699

final note - I did get this error that I had never seen before. Hard to replicate
Screenshot 2024-03-19 at 17 50 29

this was the only new thing and happened pretty rarely. I think we're good to go on this

**Comment by Da-Colon** [comment]
New Error you are experiencing is this?

https://github.com/orgs/decentdao/projects/32/views/3?pane=issue&itemId=56621736

**Pull request merged**